### PR TITLE
feat(codedb): baseline index with independent rebuild cadence

### PR DESCRIPTION
**Pull request by rsnodgrass**
## Summary

- **Baseline index infrastructure**: Adds a daemon-managed baseline CodeDB index stored in the ledger cache (`~/.local/share/sageox/<endpoint>/ledgers/<repo_id>/.sageox/cache/codedb-baseline/`), independent of per-worktree indexes
- **Decoupled rebuild cadence**: Baseline index freshness checks run on their own 15-minute `BaselineCheckInterval` tick, independent of ledger/team-context pull frequency — so tuning sync intervals doesn't accidentally starve or spam index rebuilds
- **Multi-source content tracking**: New `contentSourceFingerprint` computes a composite hash from the ledger HEAD SHA + all team context HEAD SHAs, so index rebuilds trigger when *any* content source changes (discussions, docs, SOUL.md, sessions)
- **CLI integration**: `ox code search`, `ox code sql`, and `ox code status` prefer the baseline index when available, with graceful fallback to legacy worktree index

## Architecture

```mermaid
graph TD
    A[SyncScheduler tick every 15min] --> B[contentSourceFingerprint]
    B --> C{fingerprint changed?}
    C -->|yes| D[triggerBaselineRebuild]
    C -->|no| E[skip]
    D --> F[BuildBaseline]
    F --> G[baseline index in ledger cache]
    
    H[Ledger HEAD] --> B
    I[Team Context 1 HEAD] --> B
    J[Team Context N HEAD] --> B
    
    K[ox code search] --> L{baseline exists?}
    L -->|yes| G
    L -->|no| M[legacy worktree index]
```

## Test plan

- [x] 34 new tests covering baseline lifecycle, concurrent access, error recovery, and edge cases
- [x] `TestBuildBaseline_*` — 10 tests for build/rebuild/error/concurrent scenarios
- [x] `TestBaselineStats_*` — stats reporting with/without index present
- [x] `TestContentSourceFingerprint_*` — 6 tests for composite hash edge cases (no team contexts, partial failures, ordering stability)
- [x] `TestTriggerBaselineRebuild_*` — 4 tests for trigger conditions (SHA changed, same SHA, no ledger, git failure)
- [x] `TestResolveBaselineCodeDBDir_*` — 3 tests for CLI fallback behavior
- [x] `TestBaselineCheckInterval` — config value assertion
- [x] All tests pass with `-race`; pre-existing failures unrelated

Co-authored-by: SageOx <ox@sageox.ai>

## Summary by CodeRabbit

* **New Features**
  * Baseline CodeDB indexing with periodic background rebuild checks (default 15m; 0 disables)
  * CLI code commands prefer baseline indexes when available; status/reporting reflects baseline vs worktree backend

* **Tests**
  * Extensive new tests for baseline lifecycle, rebuild triggering/fingerprinting, concurrency, and stats behavior

* **Other**
  * Indexing/status metrics now include baseline fields

* **Documentation**
  * Added testing guidance for concurrency and subsystem lifecycles

### CODE REVIEW

File: internal/daemon/sync.go [hunk 1]
```diff
@@ -1163,6 +1178,56 @@ func (s *SyncScheduler) syncFromWatcher(ctx context.Context) {
...
+	// build composite fingerprint from all content sources
+	fingerprint := s.contentSourceFingerprint(ctx, ledger.Path)
+	if fingerprint == "" || fingerprint == s.lastBaselineSha {
+		return
+	}
+
+	oldFingerprint := s.lastBaselineSha
+	s.lastBaselineSha = fingerprint
+	s.logger.Info("codedb baseline rebuild triggered", "old_fingerprint", oldFingerprint, "new_fingerprint", fingerprint)
+	go s.codedb.BuildBaseline(ctx, ledger.Path)
```
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply internal/daemon/sync.go]: Fixed.
